### ISSUE TITLE
Add share option to room's context menu

### ISSUE THREAD

**Issue by SimonBrandner** [vector-im/element-web#16457]
**Is your suggestion related to a problem? Please describe.**
I often find myself needing to share a link to a room. For that, I need to switch to it, click share, copy the link, close the dialog and switch back to the room in which I was before. That's too many clicks away...

**Describe the solution you'd like**
Add share option to room's context menu

**Additional context**
There is already [a PR](https://github.com/matrix-org/matrix-react-sdk/pull/5648) for an invite option.

**Comment by niquewoodhouse** [comment]
I think this is a good idea - the PR referred to is merged now, but the invite dialog isn't super helpful if you're just looking to copy a link, so I think this issue still has something to it.

So I wonder if the invite dialog should change to include a way a more obvious way to copy a link

Idea 1

<img width="1440" alt="invite dialog 1@2x" src="https://user-images.githubusercontent.com/1587166/112647475-79f3bd00-8e40-11eb-959a-4675cf33a201.png">


Or idea 2 - it takes all the sharing options of the share room dialog, so we can just have 1 way to share room/invite to room.

<img width="1440" alt="invite dialog 2@2x" src="https://user-images.githubusercontent.com/1587166/112647440-6d6f6480-8e40-11eb-859a-764495226c6c.png">

**Comment by SimonBrandner** [comment]
Hey @niquewoodhouse, thanks! I really like the second suggestion. Merging the two dialogs seems like a very nice thing to do

**Comment by SimonBrandner** [comment]
@niquewoodhouse, as I've mentioned above, I like the idea of merging the two dialogs. At some point, I would like to take a look into implementing that. Is that fine by the design team? If so, there is one more thing that needs to be solved, I think. What should I do with the share dialog? It's still used to share users and events. So, are there any changes to be done to that dialog and where should I keep using it and where not?

**Comment by niquewoodhouse** [comment]
Hey @SimonBrandner thanks for raising the question, I'll ask this at a design meeting this week and add a comment here later in the week.

**Comment by niquewoodhouse** [comment]
**Update:**

Sorry it took a bit longer to update here than I expected. 

Had a meeting with design and it's a mix of good and not so good news.  

**Good**

There's an agreement we _can_ merge the two dialogs, pending better designs.  There were some requests to change the design a bit, which I'm happy to do, might just take a bit of time to get around to it.  

**Not so good**

There's a bit of extra complexity in merging them as some features depend on if the room being shared is private or public.  For example, if it's a private room, sharing a link won't work.  Design is interested in exploring adding ways that would help people invite to private rooms but that probably expands up the scope a bit. Example mockup below.

<img width="1440" alt="invite dialog 5@2x" src="https://user-images.githubusercontent.com/1587166/117867228-00008180-b290-11eb-9a8a-403fa923cd55.png">

I don't know how realistic that is but I just raise this as an example of unknowns that we might face in this issue. 

I do remember @t3chguy telling me that the existing invite dialog is a bit complex at the moment - with all the things it does/can do.  So this might be a bigger issue than we realise.

>  At some point, I would like to take a look into implementing that. Is that fine by the design team? 

So yes, it's fine by the design team but I think there's more design work to do before merging them, so you might be waiting a bit.

> `What should I do with the share dialog? It's still used to share users and events. So, are there any changes to be done to that dialog and where should I keep using it and where not?`

I think the goal should be to replace all existing versions of the share dialog for consistency.

Hope that helps, pls feel free ask anything else here, thanks.

**Comment by ShadowJonathan** [comment]
@niquewoodhouse re: that mock-up, after some clarification in #element-dev, I understand the "visibility options" there will actually change the room's visibility? If so, then that's not made clear.

Coming from discord (which uses invite links) and drawing some parallels to online document collaboration, the expectation often is is that the visibility options are carried with *the link*, and doesn't widely apply to the "applicative content" (doc, chat server, etc.), if it is, that is noted explicitly.

Feedback; changing it to "room visibility" where it says these options, i.e. "anyone with a link to the room can knock/join". The crux is is that here it is said about the *link*, and not the room, while the visibility options will apply to the room.

**Comment by t3chguy** [comment]
copy link got added to the ctx menu

<img width="177" alt="image" src="https://user-images.githubusercontent.com/2403652/229776285-a668104a-8d3c-40e9-b6d3-cdbaca349198.png">
